src/commands/web/spec/updateTagSource.spec.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import path from 'path'
+import {
+  createFolder,
+  deleteFolder,
+  generateTimestamp,
+  readFile,
+  ServerType,
+  Target
+} from '@sasjs/utils'
+import { updateTagSource } from '..'
+
+describe('updateTagSource', () => {
+  let destinationPath: string
+
+  beforeAll(async () => {
+    destinationPath = path.join(
+      __dirname,
+      `cli-test-web-updateTagSource-${generateTimestamp()}`
+    )
+    await createFolder(destinationPath)
+    process.projectDir = destinationPath
+  })
+
+  afterAll(async () => {
+    await deleteFolder(destinationPath)
+  })
+
+  it(`should create program for SAS9 having base64 encoded content`, async () => {
+    const scriptTag: HTMLLinkElement = document.createElement('link')
+
+    const sourcePath = '../testFiles'
+    const target = { serverType: ServerType.Sas9 } as any as Target
+    const assetPathMap = []
+
+    const script1Filename = 'script1.js'
+    const script1SASProgramName = 'script1-js.sas'
+    const script1Base64 = `O1sn4pWQJywgJ+KVpCcsICfilZQnLCAn4pWXJywgJ+KVkCcsICfilacnLCAn4pWaJywgJ+KVnScsICfilZEnLCAn4pWfJywgJ+KUgCcsICfilLwnLCAn4pWRJywgJ+KVoicsICfilIInXQo=`
```

Comment:
it would be better to generate base64 rather than hard code it

Reply:
is it not the idea to validate the accuracy of the encoding process?  (we had issues with the way the previous library did it for certain characters)